examples/gateway2.js

```diff
@@ -0,0 +1,192 @@
+'use strict'
```

Comment:
Looks great, thanks for doing that! Could we make the file name more meaningful, such as `gateway-with-auth-on-type.js` for example?